01-getting-started/projects/final/app/src/main/java/com/raywenderlich/podplay/viewmodel/PodcastViewModel.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2022 Razeware LLC
+ *   
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *   
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *   
+ *   Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+ *   distribute, sublicense, create a derivative work, and/or sell copies of the
+ *   Software in any work that is designed, intended, or marketed for pedagogical or
+ *   instructional purposes related to programming, coding, application development,
+ *   or information technology.  Permission for such use, copying, modification,
+ *   merger, publication, distribution, sublicensing, creation of derivative works,
+ *   or sale is expressly withheld.
+ *   
+ *   This project and source code may use libraries or frameworks that are
+ *   released under various Open-Source licenses. Use of those libraries and
+ *   frameworks are governed by their own individual licenses.
+ *   
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ */
+
+package com.raywenderlich.podplay.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.raywenderlich.podplay.db.PodPlayDatabase
+import com.raywenderlich.podplay.db.PodcastDao
+import com.raywenderlich.podplay.model.Episode
+import com.raywenderlich.podplay.model.Podcast
+import com.raywenderlich.podplay.repository.PodcastRepo
+import com.raywenderlich.podplay.util.DateUtils
+import com.raywenderlich.podplay.viewmodel.SearchViewModel.PodcastSummaryViewData
+import java.util.*
+
+class PodcastViewModel(application: Application) : AndroidViewModel(application) {
```

Comment:
```
import androidx.lifecycle.ViewModel

class PodcastViewModel() : ViewModel() {
```

Reply:
Will update this. Thanks for pointing out.

Reply:
Assigned @vguzzi - https://trello.com/c/MCRXMGlr/28-use-up-to-date-viewmodel-practises

Reply:
This seems to have been used due to the DAO needing access to the application context. Upon further investigation to see how the app context can be retrieved in the ViewModel, the recommendation is to either not use the context (best practise) or use AndroidViewModel.

AndroidViewModel isn't actually deprecated, it's just a wrapper of ViewModel with access to the app context. Leaving as is to avoid restructuring.